src/objects.rs

```diff
@@ -296,6 +297,20 @@ impl Id<PhysicalMode> for PhysicalMode {
     }
 }
 
+impl Hash for PhysicalMode {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        (&self.id, &self.name).hash(state);
+    }
+}
+
+impl PartialEq for PhysicalMode {
+    fn eq(&self, other: &PhysicalMode) -> bool {
+        self.id == other.id && self.name == other.name
+    }
+}
+
+impl Eq for PhysicalMode {}
```

Comment:
Is it ok and useful to declare `impl` without any function ?

Reply:
it has to be there but i guess it uses the declaration from the trait PartialEq
```
Eq has no extra methods, it is only informing the compiler that this is an equivalence relation rather than a partial equivalence relation.
```
see https://doc.rust-lang.org/std/cmp/trait.Eq.html